src/dotnet/commands/dotnet-run/RunCommand.cs

```diff
@@ -33,6 +38,39 @@ public int Start()
 
             ICommand runCommand = GetRunCommand();
 
+            if (!NoLaunchProfile)
+            {
+                var buildPathContainer = File.Exists(Project) ? Path.GetDirectoryName(Project) : Project;
```

Comment:
I think this whole block should get encapsulated in its own method as well. Something like ApplyLaunchProfileSettings or something like that. And then it can return a command or modify the one being passed. It feels kind of loose to have this separate bit of code here like this.